patterns/1-initial/extensions-to-manage-contributions-at-scale.md

```diff
@@ -0,0 +1,81 @@
+## Title
+
+Extensions to Manage Contributions at Scale
+
+## Patlet
+
+TBD
+
+## Problem
+
+As the number of new feature contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping repositories well maintained?
+
+## Context
+
+- More employees are contributing to strategic InnerSource codebases, some of which are scaling rapidly with new feature contributions. This is adding a huge code review backlog on a smaller group of maintainers, resulting in maintainer burnout as well as slowing down release of new capabilities.
```

Comment:
maybe this one https://www.jeffgeerling.com/blog/2022/burden-open-source-maintainer?

Reply:
Optimally we would link to something specific to InnerSource. But we could use open source references as a backup as well. 

Also found theses two for open source:
https://opensource.com/article/19/11/burnout-open-source-communities
https://opensource.com/article/21/7/burnout-open-source

Reply:
I feel these articles did not describe the reasons for maintainer burnout in particular. I  could not find any external resources on how burnout applies in inner source context. I feel it could be dismissed as - "you should do innerSource during work hours, and get paid for it". But that does not solve the issue we faced. So I am getting rid of the term.


As a backstory, Our burnout was due to:
1. Full time jobs becoming about code reviews and answering questions - leading to maintainers ( who are developers ) being unhappy as they did not have any time to do any feature development themselves. This also affects the mission as maintainers did not have time to build the collaboration infrastructure needed to engage communities, or improve user experience in product . Without writing code themselves, it is hard for maintainers to set design principles or standards.
2. Lack of recognition or feeling of accomplishment - as we did not feel like we had anything to show for at the end of the day, if all we did was code reviews, and weren't able to accomplish individual goals or implement our ideas. Besides there was a large influx of incoming contributions , not all of which even get adopted, so it was also unnecessary cost for company to invest in hardening of features that did not have internal use cases. 

_(The large influx of contributions is very possible in AI domain with broad range of algorithms available and can very quickly become overwhelming. It took a few months of collaboration to harden each feature contribution to meet standards set by project)_

Extensions helped us free maintainers' time from code reviews to a large extent to work on other development as well, and results in only strategic features being ported to the primary project which have gained adoption, increasing feeling of accomplishment for time spent in hardening those features

Reply:
I think it would be worth putting your definition of maintainer burnout somewhere into this pattern, especially as I don't think we got this anywhere else in another pattern yet.

Maybe into the [Story section](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/meta/pattern-template.md#story-optional) of the pattern?

Reply:
ya I think we can add it, but I also dont think thsi pattern is necssarily tied to the story. That is, we want to actively prevent burnout based on what is described in the context and not wait till we hit burnout state. We ourselves applied the solution late and want to make it as a pattern to prevent any other team from hitting this state.

But might be worth mentioning in story as this is what could happen and we want to avoid it proactively

Reply:
Sounds good.

We can add something to the story like this:
> To confirm whether this pattern might apply to your org, please check the points listed under Context.
> This story here is meant to provide additional background ....